### PR TITLE
ci: add dependabot config for updating vendored packages

### PR DESCRIPTION
**Pull request by nixpanic**
Vendored dependencies need updating on regular basis. This is currently
done manually by developers, but it can be automated by Dependabot. By
dropping the dependabot.yml config file in the .github/ directory the
bot should get enabled.

See-also: https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/enabling-and-disabling-version-updates

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

**Review by yati1998 — COMMENTED** [review]
LGTM

**Review by humblec — DISMISSED** [review]

**Review by Rakshith-R — DISMISSED** [review]

**Review by yati1998 — DISMISSED** [review]

**Comment by nixpanic** [comment]
Marking this `DNM`, as I want to check the subject of PRs that dependabot creates.

**Review by humblec — DISMISSED** [review]

**Comment by nixpanic** [comment]
Added an additional config option to set the `rebase: ` prefix that we expect in subjects of commit messages. We might need a change to the DCO/commitlint configuration as @dependabot should not add a `Signed-off-by` line.

**Review by yati1998 — DISMISSED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+version: 2
+updates:
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (pkalever) [inline .github/dependabot.yml:7]: Why not daily?

**Comment by pkalever** [comment]
This is a cool addition :+1:

File: .github/dependabot.yml (hunk 1 above)
Comment (nixpanic) [reply .github/dependabot.yml:7]: there are lots of dependencies, not sure how how often updates are published... we might want to change this later, when we are confident it is useful

Comment (pkalever) [reply .github/dependabot.yml:7]: Sure, makesense. :+1:

**Review by pkalever — DISMISSED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (humblec) [reply .github/dependabot.yml:7]: its good to weekly as it can pull the consolidated changes together for few at time.. daily looks like a an overkill :)

Comment (pkalever) [reply .github/dependabot.yml:7]: I just want to avoid the need to trigger this manually on a release day, if that makesense.

**Comment by nixpanic** [comment]
Added a commit to use `gomodules.xyz/jsonpatch/v2@v2.2.0` from GitHub, the gomodules.xyz service doesn't do redirection correctly at the moment and that causes mod-check to fail.

**Comment by nixpanic** [comment]
It seems to be a slow Monday for repositories of Go packages :sob: 

**mod-check** continues to fail, but with an other issue:
```
running: go mod verify
go: github.com/libopenstorage/secrets@v0.0.0-20210709082113-dde442ea20ec requires
	github.com/hashicorp/vault@v1.4.2 requires
	github.com/hashicorp/vault-plugin-auth-cf@v0.5.4 requires
	github.com/cloudfoundry-community/go-cfclient@v0.0.0-20190201205600-f136f9222381 requires
	code.cloudfoundry.org/gofileutils@v0.0.0-20170111115228-4d0c80011a0f: unrecognized import path "code.cloudfoundry.org/gofileutils": https fetch: Get "https://code.cloudfoundry.org/gofileutils?go-get=1": x509: certificate signed by unknown authority
```

**Comment by Rakshith-R** [comment]
> /retest all

running ci tests since this pr also modifies `go.mod` now, just to make sure everything still works :wink: 
Otherwise lgtm
Thanks!

**Comment by Rakshith-R** [comment]
> /retest ci/centos/mini-e2e/k8s-1.19

failed due to #2264

**Comment by nixpanic** [comment]
> > /retest all
> 
> running ci tests since this pr also modifies `go.mod` now, just to make sure everything still works wink
> Otherwise lgtm
> Thanks!

I do not think this is needed, the contents of the .go files under `vendor/` have not been changed, even the hashes of the replacement locations are the same.

**Review by Rakshith-R — APPROVED** [review]
I missed the vendor folder part 😅🤦.
Thanks!

**Review by humblec — APPROVED** [review]

**Pull request merged**